hack/make/dependencies.make

```diff
@@ -1,4 +1,4 @@
-##@ Build Dependencies
+## Build Dependencies
```

Comment:
This is used to add a group in the help when running `make help` see https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/Makefile#L40

Reply:
ack. i'll add it back in. 👍